src/main/java/com/blackduck/integration/detect/workflow/project/ProjectNameVersionDecider.java

```diff
@@ -37,7 +37,7 @@ public NameVersion decideProjectNameVersion(List<DetectTool> preferredDetectTool
         String decidedProjectVersionName;
         if (StringUtils.isNotBlank(projectVersionOptions.overrideProjectVersionName)) {
             decidedProjectVersionName = projectVersionOptions.overrideProjectVersionName;
-        } else if (chosenToolVersion.isPresent() && !chosenToolVersion.get().isBlank()) {
+        } else if (chosenToolVersion.isPresent() && !chosenToolVersion.get().trim().isEmpty()) {
```

Comment:
Can this be null?

Reply:
In reality I think probably not given the way these objects are constructed but...that's probably famous last words so let's protect it. Added an additional check.

Reply:
Some food for thought: the whole point of `Optional`, I believe, is to make null-checking look cleaner. By adding a raw null check we're undoing that benefit and are making these checks even less pretty.